### PR TITLE
Allow frontend to cache and option to purge

### PR DESCRIPTION
**Pull request by wilson1000**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -197,5 +198,6 @@ services:
       target: build-cron
     environment:
       NGINX_HOST: ${NGINX_SERVICE_PORT}
+      NGINX_SERVICE_PORT: ${NGINX_SERVICE_PORT}
```
Comment (EarthlingDavey) [inline docker-compose.yml:201]: Do we need to add `NGINX_SERVICE_PORT: "http://localhost:8080"` to `deploy/(demo|development|staging)/config.yml`

File: deploy/config/local/nginx/server.conf [hunk 2]
```diff
@@ -75,16 +122,23 @@ server {
         deny all;
     }
 
+    # Deny access to load-scripts.php and load-styles.php to prevent DoS attacks.
```
Comment (EarthlingDavey) [inline deploy/config/local/nginx/server.conf:125]: Nice, I think this block should also be in `deploy/config/server.conf`

**Review by EarthlingDavey — APPROVED** [review]
Built and runs locally 💯

File: docker-compose.yml (hunk 1 above)
Comment (wilson1000) [reply docker-compose.yml:201]: This was to solve a local problem where the NGINX_SERVICE_PORT (that is present in K8S) is available for the CRON to work

Comment (wilson1000) [reply docker-compose.yml:201]: So basically, it is already in the environment 😄

**Comment by EarthlingDavey** [comment]
See: https://github.com/ministryofjustice/intranet/blob/044aa597d8422b3f6c5747d6310d696811704d1f/docker/conf/nginx/server.conf#L15 

Having `$cookie_dw_agency` in the cache key will fix the issue. It will create individual caches for each agency.

Purging the cache might need a bit of work.

**Comment by EarthlingDavey** [comment]
If we do:

`fastcgi_cache_key "$real_scheme$request_method$host$request_uri$cookie_dw_agency"; `

with $cookie_dw_agency at the end.

We can add a wildcard to the end of the purge line:

`fastcgi_cache_purge mojiCache "$real_scheme$request_method$host$1*";`

to purge a page for all agencies.

> If the [cache key](https://nginx.org/en/docs/http/ngx_http_fastcgi_module.html#fastcgi_cache_key) of a purge request ends with an asterisk (“*”), all cache entries matching the wildcard key will be removed from the cache.

https://nginx.org/en/docs/http/ngx_http_fastcgi_module.html

**Comment by EarthlingDavey** [comment]
While we're here, please could we also add info.php and errors.php onto the exclude list?

@wilson1000 if you're focusing on the other PR and (agree with my suggestions)... I don't mind adding these handful of changes.

**Comment by EarthlingDavey** [comment]
There's one minor improvement that could be made.

The way I suggested to use `*` at the end of `fastcgi_cache_purge`, means that if:

1. We have cache for http://intranet.docker/guidance/ & http://intranet.docker/guidance/fire-health-safety/
2. We clear cache for guidance by hitting `http://intranet.docker/purge-cache/guidance/`

Then the cache for fire-health-safety will be purged, because the * not only matching the cookie value, but any path that's following `guidance`.

To address this, we can add a colon before the cookie value: e.g.
`fastcgi_cache_key ... $request_uri:$cookie_dw_agency";`
`fastcgi_cache_purge ... $host$1:*";`

I've tested this locally, and with the above scenario, `fire-health-safety` is not purged.

**Pull request merged**